src/dmd/lexer.d

```diff
@@ -2647,24 +2647,26 @@ private struct TimeStampInfo
 
     public static void initialize(const ref Loc loc) nothrow
     {
-        if (initdone)
+        if (cast()initdone)
             return;
 
-        initdone = true;
-        time_t ct;
-        // https://issues.dlang.org/show_bug.cgi?id=20444
-        if (auto p = getenv("SOURCE_DATE_EPOCH"))
+        if (cas(&initdone, false, true))
```

Comment:
This still allows other threads to read the timestamp before it's fully initialized?

Reply:
yes ... indeed. 
other threads would have to loop until initdone is true.

Reply:
Fixed now.